File: patient-ms/src/main/java/com/medical/assessment/patientms/persistence/entity/Patient.java
package com.medical.assessment.patientms.persistence.entity;

import com.medical.assessment.patientms.persistence.enums.Gender;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Builder
@Data
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "patient")
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    @Column(name = "patient_id",nullable = false)
    private Long patientId;

    @Column(name = "last_name", nullable = false, length = 100)
    private String lastName;

    @Column(name = "first_name", nullable = false, length = 100)
    private String firstName;

    @Column(name = "birth_date", nullable = false)
    private Instant birthDate;

    @Enumerated(EnumType.STRING)
    @Column(name = "gender", nullable = false, length = 10)
    private Gender gender;

    @Column(name = "phone_number", length = 20)
    private String phoneNumber;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "address_id",unique = true)
    private Address address;
}

File: frontend-ms/src/types/Patient.ts
export interface Patient{
    id: number;
    firstName: string;
    lastName: string;
    birthDate: string;
    gender: 'M' | 'F';
    address: string;
    phoneNumber: string;
}

export interface PatientCreateDto {
    firstName: string;
    lastName: string;
    birthDate: string;
    gender: 'M' | 'F';
    address?: string;
    phoneNumber?: string;
}

export interface PatientUpdateDto {
    firstName?: string;
    lastName?: string;
    birthDate?: string;
    gender?: 'M' | 'F';
    address?: string | null;
    phoneNumber?: string | null;
}